[t-kandasami] iP

t-kandasami opened the pull request.

t-kandasami committed:
Rename chatbot name to Prime. Add simple greeting and exit.

t-kandasami committed:
Add *.class files into the .gitigore

t-kandasami committed:
Update Prime bot to echo user's input and gracefully exit once the user types command bye

t-kandasami committed:
Update the logo into ASCII Block Art Style

t-kandasami committed:
Add Task Class to represent tasks

t-kandasami committed:
Update the bot to add anything enter to be added into task list and list command

t-kandasami committed:
Update Task.java toString() overloaded method

t-kandasami committed:
Update Prime.java to add markTask and unmarkTask functionality to bot

XuHh03 commented on src/main/java/Prime.java:
```diff
@@ -34,10 +33,38 @@ private static void listTasks() {
         printIndented("Here are your tasks in your list: ");
         for (int i = 0; i < taskCounter; i++) {
-            printIndented((i + 1) + ". " + tasksList[i].toString());
+            printIndented((i + 1) + "." + tasksList[i].toString());
```
I like how you use a method instead of repetitive typing

t-kandasami replied:
Thanks

V1T0bh commented on src/main/java/Prime.java:
```diff
@@ -1,13 +1,43 @@
     private static final String AGENT_NAME = "Prime";
+    private static final int MAX_TASKS = 100;
+    private static final Task[] tasksList = new Task[MAX_TASKS];
+    private static int taskCounter = 0;
```
good practice. constants declared instead of hardcoding to specfic lines for consistency.

t-kandasami replied:
👍

V1T0bh commented on src/main/java/Prime.java:
```diff
@@ -1,13 +1,43 @@
+            taskCounter++;
+            printIndented("added: " + description);
+        } else {
+            printIndented("Task list is full Human! Maybe its time you finish some tasks.");
```
FUNNY

V1T0bh commented on src/main/java/Prime.java:
```diff
@@ -26,11 +56,21 @@ public static void main(String[] args) {
+
+            if (!userInput.equalsIgnoreCase("bye")) {
                 printIndented(LINE_BREAK);
             }
```
`printIndented(LINE_BREAK);` could have been added to case OR check for "bye" and then break before entering swithc statement

t-kandasami replied:
That's a good suggestion. Planning to print line break inside case in my next few refactors.

V1T0bh commented on src/main/java/Task.java:
```diff
@@ -21,6 +21,12 @@ public void setAsNotDone() {
     public String toString() {
-        return (getDescription());
+        String output;
+        if (isDone) {
+            output = "[X] " + description;
+        } else {
+            output = "[ ] " + description;
+        }
```
try using a **ternary operator**, you can use that to shorten your code

t-kandasami replied:
I felt like it would be better to make the code more obvious by using if else

V1T0bh reviewed (commented):
Overall code looks great! I like the logo you made for PRIME. Keep up the good work!

t-kandasami committed:
Add Deadline Children Class

t-kandasami committed:
Add Event Children Class

t-kandasami committed:
Add ToDo Children Class

t-kandasami committed:
Update Prime.java class methods

t-kandasami committed:
Update Prime.java

* Added the creation Deadline, ToDo, Event functionalities.
* Created parsing methods to ensure the proper retrieval of proper informaion for each type of Task.
* Refactored unmark and mark functions to use a single function called toggleMarkTask

xylonc commented on src/main/java/Prime.java:
```diff
@@ -0,0 +1,180 @@
+
+public class Prime {
+    // Constants
+    private static final String LINE_BREAK = "____________________________________________________________";
```
can try alternative of 
System.out.println("\n" + "-".repeat(30));

xylonc commented on src/main/java/Prime.java:
```diff
@@ -0,0 +1,180 @@
+    // Constants
+    private static final String LINE_BREAK = "____________________________________________________________";
+    private static final String AGENT_NAME = "Prime";
+    private static final int MAX_TASKS = 100;
```
nice job capitalising the constants!

t-kandasami replied:
thanks

xylonc commented on src/main/java/Prime.java:
```diff
@@ -0,0 +1,180 @@
+    }
+
+    private static void addDeadline (String description, String by) {
+        if (taskCounter < MAX_TASKS && description != null && by != null) {
```
nice job considering the edge cases

xylonc commented on src/main/java/Prime.java:
```diff
@@ -0,0 +1,180 @@
+            userInput = scanner.nextLine().trim().toLowerCase();
+            printIndented(LINE_BREAK);
+
+            if (userInput.startsWith("mark ")) {
```
good job limiting arrowhead style code with uses of else if

xylonc commented on src/main/java/Prime.java:
```diff
@@ -0,0 +1,180 @@
+    }
+
+    public static void main(String[] args) {
+        String logo = """
```
amazing logo

xylonc commented on src/main/java/Prime.java:
```diff
@@ -0,0 +1,180 @@
+    private static String parseFrom(String command) {
+        int indexOfFrom = command.indexOf("/from");
+        int indexOfTo = command.indexOf("/to");
+        return command.substring(indexOfFrom + "/from".length(), indexOfTo).trim();
```
great uses of trim to not have to deal with white spaces and use of "/form".length to remove human error of counting

t-kandasami committed:
Test: Add the input commands in input.txt and expected output in expected.txt

t-kandasami committed:
Reformat: run the auto reformater in all class files to ensure the coding conventions

t-kandasami committed:
Refactor: Create a UserInterface.java
Using the UserInterface Class to manages all the printing methods of the chatbot

t-kandasami committed:
Refactor: Add CommandType.java
I created a CommandType Enums so I can use this to switch case the command later.

t-kandasami committed:
Add Command.java

t-kandasami committed:
Add Parser.java
Took out my parsing methods from Prime.java and put them in another class. Used better and appropriate String Methogs

t-kandasami committed:
Update: Create a new method to print the tick icon.
this update the other related classes

t-kandasami committed:
Create TaskManager: Inspired by Prof lecture.
I created a class to manage the task opperations

t-kandasami committed:
Update Prime.java

* Change my chat bot to use the other abstracted classes

t-kandasami committed:
Update UI Test Expected.txt

t-kandasami committed:
Fix: Print the 4 spaces for the line break as well

t-kandasami committed:
Task.java: Made Task class abstract

t-kandasami committed:
Add PrimeException and its subclasses for specific errors

t-kandasami committed:
Made changes to the Excepected.txt

t-kandasami committed:
Change to TaskManager and Prime

* Updated my if-else error messages to throw exception where necessary

t-kandasami committed:
Merge branch 'branch-Level-5' into master

t-kandasami committed:
Create the prime.exceptions package

t-kandasami committed:
Create the prime.manager package

t-kandasami committed:
Create prime.parser package

t-kandasami committed:
Create prime.task package

t-kandasami committed:
Create prime.ui package

t-kandasami committed:
Create the prime package

t-kandasami committed:
Merge branch 'branch-A-Packages' into master

t-kandasami committed:
Change Task[] tasks to ArrayList<Task> task

* Using ArrayList to store the tasks. So that I can use builtin methods

t-kandasami committed:
Update textUi-testing batch script

* update the script to accomdate the creation of packages

t-kandasami committed:
Add delete functionality to PrimeBot

JoanneHing commented on src/main/java/prime/Prime.java:
```diff
@@ -0,0 +1,96 @@
+                    break;
+                default:
+                    throw new InvalidCommandException();
+                }
```
Good job on using switch case here instead of if-else blocks

JoanneHing commented on src/main/java/prime/exceptions/EmptyByException.java:
```diff
@@ -0,0 +1,9 @@
+
+public class EmptyByException extends PrimeException {
+    public EmptyByException(CommandType commandType) {
+        super("OOPS!!! The Task Number " + commandType.toString() + " is invalid.");
```
Good job on using a customized input of the message string related to the exception class.

JoanneHing commented on src/main/java/prime/manager/TaskManager.java:
```diff
@@ -0,0 +1,49 @@
+import prime.ui.UserInterface;
+
+public class TaskManager {
+    private static final int MAX_TASKS = 100;
```
Good job on avoiding magic numbers here.

JoanneHing commented on src/main/java/prime/parser/CommandType.java:
```diff
@@ -0,0 +1,13 @@
+package prime.parser;
+
+public enum CommandType {
+    BYE,
+    LIST,
+    MARK,
+    UNMARK,
+    TODO,
+    DEADLINE,
+    EVENT,
+    INVALID
+}
```
Good to see you trying out on enums!

JoanneHing commented on src/main/java/prime/ui/UserInterface.java:
```diff
@@ -0,0 +1,26 @@
+package prime.ui;
+
+public class UserInterface {
+    private static final String LINE_BREAK = "____________________________________________________________";
```
Good that you extracted out this repetitive String out as constant.

JoanneHing reviewed (commented):
Overall very good code done. Good to see you followed the coding standards and keeping up well with the progress.

t-kandasami committed:
Update .gitignore to not track the data directory

t-kandasami committed:
Create Storage.class

* load() - Load Tasks (Task ArrayList) from file
* save(task) Save task into the file
* parseTask(line) Convert line from file into a Task Object
* formatTask(task) Convert a Task Object into a String

t-kandasami committed:
Create new getter access methods for the class

t-kandasami committed:
Update TaskManager to handle the Storage methods

t-kandasami committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/prime/manager/TaskManager.java

t-kandasami committed:
Fix: Update the deleteTask to update file

t-kandasami committed:
Bug fixed: Solve the Empty To, From, By Exeception

t-kandasami committed:
Refactor: Create Command and Subclasses

t-kandasami committed:
Bug Fix: Change Toggle Task to Mark and Unmark

t-kandasami committed:
Change Parser to use Command and Subclasses

t-kandasami committed:
Refactor: Change Prime to use command.execute()

t-kandasami committed:
Merge branch 'A-MoreOOP'

t-kandasami committed:
Create FindCommand

t-kandasami committed:
Change TaskManager to allow FindTasks

t-kandasami committed:
Reformat: Used AutoReformater IDE

t-kandasami committed:
Merge pull request #1 from t-kandasami/branch-Level-9

Branch level 9 - Supporting Find Command

t-kandasami committed:
A-JavaDoc: Add JavaDoc

t-kandasami committed:
Merge pull request #2 from t-kandasami/A-JavaDoc

A-JavaDoc: Add JavaDoc

t-kandasami committed:
Docs: Add User Guide Documentation

t-kandasami committed:
Merge pull request #3 from t-kandasami/A-UserGuide

Docs: Add User Guide Documentation

t-kandasami committed:
Reformat code
